### ISSUE TITLE
[Question] Seeding files (Server-side) on multiple servers, with control.

### ISSUE THREAD

**Issue by ghost** [webtorrent/webtorrent#615]
I'm planning on having a few servers built specifically for seeding files. I'd also like to be able to have control over who gets to seed/leech from the torrents I'm hosting on top of that. Is there a specific method for doing such, or any examples? 

I am hoping for something that would look like this:
`server.seed(file, function(torrent) {

```
     torrent.leech(function(client) {});

     torrent.seed(function(client) {});

});
```

`

in a sense, where I can see who's doing what with the torrent (so I can block them from accessing the web server I'll be hosting if they abuse).

tldr; What would be the recommended way to create a torrent, and have other servers seeding it.
And have the initial server that created the torrent managing the clients (to prevent abuse) seeding/leeching the torrent.

**Comment by Stiveknx** [comment]
A few months ago I started a project similar to what you're talking.
https://github.com/Stiveknx/webtorrent-app-remote

It was writen in node, it uses Loki database.
I didn't have time to continue the project, and I stoped exactly where you mentioned about users.
Right now it creates new users, and allow them to create torrents or download (and seed) torrents.
What you can do it's create a track of which torrents the user created and seeded.

If you're interested in maintain it ...

**Comment by bertofer** [comment]
Hi,

You could consider [webseeding](http://www.bittorrent.org/beps/bep_0019.html) if you are using a server. You should be able to build  a REST Api around that.

I started a project with and admin panel to upload files, and then a downloads panel that enables to download the previously uploaded files via webtorrent, including webseeding. It's not finished nor in development, but if helps you 
- https://github.com/bertofer/filelibrary-admin 
- https://github.com/bertofer/filelibrary-downloads

You will probably also need the [tracker](https://github.com/feross/bittorrent-tracker), a private one. You can have tracker and RESTApi / WebSeed server sharing tokens, or things like that. I think you could achieve what you are looking for playing with those elements.
